src/components/Editor/index.js

```diff
@@ -120,21 +292,27 @@ export default class Editor extends Component {
     }
 
     render() {
-        var myWork;
         var myStepsList;
         var problemHeaderTitle = this.state.title;
         if (this.id != "newEditor") {
-            myWork = <MyWork
+            myStepsList = <MyStepsList
```

Comment:
Are all those callbacks needed? For example logic for the save and discard buttons could be in the child components and this way you could just pass this.props.history as a param here, Or even this.props.history.goBack() as a function.